python/cudf/cudf/core/udf/scalar_function.py

```diff
@@ -0,0 +1,84 @@
+from numba import cuda
+from numba.np import numpy_support
+
+from cudf.core.udf.api import Masked, pack_return
+from cudf.core.udf.templates import (
+    masked_input_initializer_template,
+    scalar_kernel_template,
+    unmasked_input_initializer_template,
+)
+from cudf.core.udf.typing import MaskedType
+from cudf.core.udf.utils import (
+    _construct_signature,
+    _get_kernel,
+    _get_udf_return_type,
+    _mask_get,
+)
+
+
+def _scalar_kernel_from_template(sr, args):
+    """
+    Function to write numba kernels for `Series.apply` as a string.
+    Workaround until numba supports functions that use `*args`
+
+    `Series.apply` expects functions of a single variable and possibly
+    one or more constants, such as:
+
+    def f(x, c, k):
+        return (x + c) / k
+
+    where the `x` are meant to be the values of the series. Since there
+    can be only one column, the only thing that varies in the kinds of
+    kernels that we want is the number of extra_args
+
+    def _kernel(retval,
+                size,
+                input_col_0,
+                offset_0,
+                extra_arg_0, # the extra arg `c`
+                extra_arg_1, # the extra arg `k`
+    ):
+        i = cuda.grid(1)
+        ret_data_arr, ret_mask_arr = retval
+
+        if i < size:
+            d_0, m_0 = input_col_0
+            masked_0 = Masked(d_0[i], _mask_get(m_0, i + offset_0)
+
+            ret = f_(masked_0, extra_arg_0, extra_arg_1)
+
+            ret_masked = pack_return(ret)
+            ret_data_arr[i] = ret_masked.value
+            ret_mask_arr[i] = ret_masked.valid
+
```

Comment:
This feels unnecessary when you can point to the templates file.

Reply:
Would you be ok with me moving these to some comments in `templates.py`? Perhaps it doesn't belong here but I still find it helpful to see an example of the kinds of kernels that are being generated. It makes it a little easier to conceptualize what is going on when debugging.

Reply:
I'd be fine with that. It's up to you where you think it's most helpful and whether you think it's useful to show the full resulting kernel vs just the parts that are formatted into the template. I'm just hesitant to include so much extra text that will have to be updated if we ever change the kernel templates (and we'll probably forget to update these).

Reply:
I ended up just shortening / removing these, the templates are clear enough by themselves.